src/main/java/io/zold/api/Copies.java

```diff
@@ -24,37 +24,78 @@
 package io.zold.api;
 
 import io.zold.api.Copies.Copy;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.cactoos.collection.CollectionOf;
 import org.cactoos.iterable.IterableEnvelope;
 import org.cactoos.iterable.IterableOf;
+import org.cactoos.iterable.Joined;
 import org.cactoos.iterable.Mapped;
 
 /**
  * Multiple copies of a Wallet.
- *
  * @since 1.0
- * @todo #41:30min The constructor of Copies should be implemented
- *  to retrieve the Wallet with the provided id in the provided remotes and
- *  group them by equivalent wallet (i.e., equivalent content) along
- *  with their remotes as explained in the whitepaper. A unit test should be
- *  added to validate this behaviour. The unit test
- *  NetworkTest.pullIsNotYetImplemented() should also be removed and replaced
- *  with a real test.
  */
 public final class Copies extends IterableEnvelope<Copy> {
 
     /**
      * Ctor.
-     *
      * @param id Id of the wallet to pull.
      * @param remotes Remote nodes.
      */
     Copies(final long id, final Iterable<Remote> remotes) {
-        super(() -> new IterableOf<>(new Copy(new Wallet.Fake(id), remotes)));
+        super(() -> copies(id, remotes));
+    }
+
+    /**
+     * Builds copies from remotes.
+     * @param id Wallet's id
+     * @param remotes List of remotes
+     * @return Iterable Iterable of Copy
+     * @throws IOException If fails
+     */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    private static Iterable<Copy> copies(final long id,
+        final Iterable<Remote> remotes) throws IOException {
+        final List<Copy> copies = new ArrayList<>(0);
+        for (final Remote remote : remotes) {
+            final Wallet wallet = remote.pull(id);
+            boolean updated = false;
+            for (int idx = 0; idx < copies.size(); idx += 1) {
+                final Copy copy = copies.get(idx);
+                if (Copies.equalWallets(copies.get(idx).wallet(), wallet)) {
+                    copies.set(idx, copy.with(remote));
+                    updated = true;
+                }
+            }
+            if (!updated) {
+                copies.add(new Copy(wallet, remote));
+            }
+        }
+        return new IterableOf<>(copies);
+    }
+
+    /**
+     * Checks if content of two wallets is equal.
+     * @param first First wallet
+     * @param second Second wallet
+     * @return Boolean Boolean
+     * @throws IOException If fails
+     * @todo #56:30min Comparing of two wallets should be done in a different
+     *  manner. Wallets should implement Bytes interface from cactoos and
```

Comment:
@Vatavuk why `Bytes`? any explanation here would be great.

Reply:
@carlosmiranda my idea was to use Equality class from cactoos which compares byte representations of two objects. This way we don't need to explicitly compare object content parameter by parameter. Wallet just need to print itself to a byte array and Equality class will do the comparison. Wallet will need to implement `Bytes` because `Equality` operates on that interface. 

More about this idea [here](https://www.yegor256.com/2017/07/11/how-to-redesign-equals.html)

Reply:
@Vatavuk sorry, but I disagree with this solution:

* I don't think we have a business need to serialize wallets into bytes
* I disagree with Yegor on this one. I think it goes against the very idea of having "living objects". I can't reconcile having "living objects" turn into "dead byte streams". And like I mentioned in my previous point, I don't yet see any connection between wallets and streams of bytes.

Reply:
@llorllale Ok, I understand. My intention was to avoid writing multiple if statements. I've updated  todo description